src/django/api/views/v1/moderation_events.py

```diff
@@ -112,43 +113,60 @@ def partial_update(self, request, pk=None):
             partial=True
         )
 
-        if serializer.is_valid():
-            serializer.save()
-            return Response(serializer.data, status=status.HTTP_200_OK)
+        if not serializer.is_valid():
+            raise ValidationError({
+                "detail": APIV1CommonErrorMessages.COMMON_REQ_BODY_ERROR,
+                "errors": [serializer.errors]
+            })
 
-        return Response(
-            {
-                "detail": 'The request body contains '
-                'invalid or missing fields.',
-                "error": [serializer.errors]
-            },
-            status=status.HTTP_400_BAD_REQUEST
-        )
+        serializer.save()
+
+        return Response(serializer.data, status=status.HTTP_200_OK)
 
     @handle_errors_decorator
-    @action(detail=True, methods=['POST'], url_path='production-locations')
-    def add_production_location(self, request, pk=None):
+    @action(
+        detail=True, methods=['POST', 'PATCH'], url_path='production-locations'
+    )
```

Comment:
Nevertheless, I decided to change this method in a different way to adhere to the Single Responsibility Principle.

Reply:
@mazursasha1990 maybe it's a good idea to leave this comment in the code:

`DRF resolves actions based on their url_path and methods. Since both add_production_location and update_production_location have the same url_path, DRF gets confused about which action to invoke which leads to errors.`